packages/web3-providers/src/rss3/index.ts

```diff
@@ -110,3 +106,22 @@ export class RSS3API implements RSS3BaseAPI.Provider, NonFungibleTokenAPI.Provid
         return createPageable(data, createIndicator(indicator))
     }
 }
+
+const getIdFromDonationURL = (url?: string) => {
+    if (!url) return
+    return url?.match(/(?<=https:\/\/gitcoin.co\/grants)\d+/g)?.[0]
+}
+
+const createCollection = (collectionResponse: RSS3BaseAPI.CollectionResponse[]): RSS3BaseAPI.Collection[] => {
+    return collectionResponse.map((collection: RSS3BaseAPI.CollectionResponse) => {
+        return {
+            ...collection,
+            title: collection?.actions?.[0]?.metadata?.title,
```

Comment:
Maybe u can declare the  `firstAction` to avoid duplicate code

`const firstActrion = first(collection?.actions)`